amanuensis/resources/admin/admin_associated_user.py

```diff
@@ -25,10 +29,73 @@ def add_associated_users(users):
     with flask.current_app.db.session as session:
         associated_user_schema = AssociatedUserSchema(many=True)
         ret = []
-        for user in users:
-            ret.append(udm.add_associated_user(session, user["project_id"], user["email"] if "email" in user else None, user["id"] if "id" in user else None))
-        associated_user_schema.dump(ret)
-        return ret
+        users_with_email = [user for user in users if "email" in user]
+        users_with_id = [
+            user
+            for user in users
+            if "id" in user
+        ]
+        if not (users_with_email or users_with_id):
+            raise UserError("The associated user must have at least an email or id. Neither was found.")
```

Comment:
This logic doesn't cover for instance the case where a user has no email and no ID, but there is at least another user with both of them.

Reply:
If the user has no email and no ID should't it raise an error or be ignored? The only other parameter is the project_id.

Reply:
raise an error.